src/azure-cli/azure/cli/command_modules/storage/operations/account.py

```diff
@@ -70,8 +73,30 @@ def create_storage_account(cmd, resource_group_name, account_name, sku=None, loc
 
     if custom_domain:
         params.custom_domain = CustomDomain(name=custom_domain, use_sub_domain=None)
+
+    # Encryption
     if encryption_services:
         params.encryption = Encryption(services=encryption_services)
+
+    if encryption_key_source is not None:
```

Comment:
```suggestion
    if encryption_key_source is not None and params.encryption:
```
To avoid `params.encryption.key_source = encryption_key_source` crush when `params.encryption` is `None`